modules/tinymce/src/core/main/ts/api/dom/DOMUtils.ts

```diff
@@ -33,119 +33,18 @@ import DomTreeWalker from './TreeWalker';
 const each = Tools.each;
 const grep = Tools.grep;
 
-interface AttrHooks {
-  style: {
-    set (elm: HTMLElement, value: string | {} | null): void;
-    get (elm: HTMLElement): string;
-  };
-  href?: {
-    set (elm: HTMLElement, value: string | null, name: string): void;
-    get (elm: HTMLElement, name: string): string;
-  };
-  src?: {
-    set (elm: HTMLElement, value: string | null, name: string): void;
-    get (elm: HTMLElement, name: string): string;
-  };
-  [key: string]: {
-    set (elm: HTMLElement, value: string | {} | null, name: string): void;
-    get (elm: HTMLElement, name: string): string;
-  };
-}
-
 const internalStyleName = 'data-mce-style';
+const numericalCssMap = Tools.makeMap('fill-opacity font-weight line-height opacity orphans widows z-index zoom', ' ');
 
-const legacySetAttribute = (elm: SugarElement<Element>, name: string, value: string | number | boolean | null) => {
-  if (Type.isNullable(value) || value === '') {
-    Attribute.remove(elm, name);
-  } else {
-    Attribute.set(elm, name, value);
-  }
-};
-
-const setupAttrHooks = (styles: Styles, settings: Partial<DOMUtilsSettings>, getContext: () => any): AttrHooks => {
-  const keepValues = settings.keep_values;
-  const keepUrlHook = {
-    set: (elm: HTMLElement, value: string | null, name: string) => {
-      const sugarElm = SugarElement.fromDom(elm);
-      if (Type.isFunction(settings.url_converter) && Type.isNonNullable(value)) {
-        value = settings.url_converter.call(settings.url_converter_scope || getContext(), value, name, elm);
-      }
-
-      const internalName = 'data-mce-' + name;
-      legacySetAttribute(sugarElm, internalName, value);
-      legacySetAttribute(sugarElm, name, value);
-    },
-
-    get: (elm: HTMLElement, name: string) => {
-      const sugarElm = SugarElement.fromDom(elm);
-      return Attribute.get(sugarElm, 'data-mce-' + name) || Attribute.get(sugarElm, name);
-    }
-  };
-
-  const attrHooks: AttrHooks = {
-    style: {
-      set: (elm, value: string | {} | null) => {
-        const sugarElm = SugarElement.fromDom(elm);
-        if (Type.isObject(value)) {
-          Css.setAll(sugarElm, value as Record<string, string>);
-          return;
-        }
```

Comment:
Removed this in the logic below because according to the types and my own debugging this will never be an object.